test/classes/sequelize-service.test.ts

```diff
@@ -612,4 +613,57 @@ describe('SequelizeService', function () {
       afterUpdateStub.restore();
     });
   });
+
+  describe('#upsert()', function () {
+    it('should create object', async () => {
+      const properties = { email: 'foo', password: 'bar' };
+      expect(await userService.count({})).to.equal(0);
+      const user = await userService.upsert(properties);
+      expect(await userService.count({})).to.equal(1);
+      expect(user).to.containSubset(properties);
+    });
+    it('should create object and compute properties', async () => {
+      const user = await userService.upsert({ email: 'foo', password: 'bar' }, { compute: ['age']});
+      expect(user).to.include.keys(['id', 'email', 'password', 'age']);
+    });
+    it('should update object', async () => {
+      const createProperties = { email: 'foo', password: 'bar', first_name: 'alice' };
+      const upsertProperties = { email: 'foo', password: 'baz', first_name: 'bob' };
+
+      expect(await userService.count({})).to.equal(0);
+      await userService.create(createProperties);
+      expect(await userService.count({})).to.equal(1);
+      const user = await userService.upsert(upsertProperties);
+      expect(await userService.count({})).to.equal(1);
+      expect(user).to.containSubset({ ...createProperties, ...upsertProperties });
+    });
+    it('should update object (with fields)', async () => {
+      const createProperties = { email: 'foo', password: 'bar', first_name: 'alice' };
+      const upsertProperties = { email: 'foo', password: 'baz', first_name: 'bob' };
+      expect(await userService.count({})).to.equal(0);
+      await userService.create(createProperties);
+      expect(await userService.count({})).to.equal(1);
+      const user = await userService.upsert(upsertProperties, { fields: ['password'] });
```

Comment:
Without the fields sequelize produces a query like this:
```MySQL
INSERT INTO `Users` (`email`,`first_name`,`password`,`date_of_birth`,`lucky_number`,`password_last_updated_at`,`created_at`,`updated_at`) VALUES ('foo','bob','baz',NULL,0,'2019-03-14 00:28:48','2019-03-14 00:28:48','2019-03-14 00:28:48') ON DUPLICATE KEY UPDATE `email`=VALUES(`email`), `password`=VALUES(`password`), `first_name`=VALUES(`first_name`), `updated_at`=VALUES(`updated_at`)
```
With `fields` set, it generates: 
```MySQL
INSERT INTO `Users` (`email`,`first_name`,`password`,`date_of_birth`,`lucky_number`,`password_last_updated_at`,`created_at`,`updated_at`) VALUES ('foo','bob','baz',NULL,0,'2019-03-14 00:28:48','2019-03-14 00:28:48','2019-03-14 00:28:48') ON DUPLICATE KEY UPDATE `password`=VALUES(`password`), `updated_at`=VALUES(`updated_at`);
```

We also use only the attributes defined in `fields` to find the row after performing the upsert (because sequelize doesn't return us the row, this is a little unsafe 🙈 )